plugins/web-terminal/src/components/TerminalComponent/TerminalComponent.tsx

```diff
@@ -65,39 +92,61 @@ export const TerminalComponent = () => {
 
   const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
-    setFormData({
-      token: tokenRef.current?.value,
-    });
+    setUsedToken(tokenRef.current?.value);
   };
 
-  const setupPod = async (link: string, token: string) => {
-    const terminalID = await createWorkspace(link, token);
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setUsedNamespace(event.target.value as string);
+  };
+  const setupPod = async (link: string, token: string, namespace: string) => {
+    let terminalID;
+    try {
+      terminalID = await createWorkspace(link, token, namespace);
+    } catch (error) {
+      return undefined;
+    }
     let workspaceID;
     let phase;
     while (phase !== 'Running') {
-      [workspaceID, phase] = await getWorkspace(link, token, terminalID);
+      [workspaceID, phase] = await getWorkspace(
+        link,
+        token,
+        terminalID,
+        namespace,
+      );
     }
     return { workspaceID, terminalID };
   };
 
   React.useEffect(() => {
-    if (!formData.token || !cluster) {
+    if (!usedToken || !cluster) {
       return;
     }
-    const { token } = formData;
     setLoading(true);
-    setupPod(cluster, token).then(names => {
+    setupPod(cluster, usedToken, usedNamespace!).then(names => {
+      if (!names) {
+        getNamespaces(cluster, usedToken).then(namespacesCluster => {
+          setNamespaces(namespacesCluster);
+          setLoading(false);
+          setUsedNamespace('');
+          setDisplayModal(true);
+        });
+        return;
```

Comment:
This way the UI always depends on loading list of all namespaces for a user, before even attempting to instantiate it in `openshift-terminal`. This should happen **only** if the first attempt in the default namespace fails.

Reply:
Should now be resolved flow should be:
Calls to create a devworkspace are triggered -> if success procceed with setuping connection to `webterminal-proxy`
if they don't succeed the modal is enabled and inside modal component call is triggered to get all namespaces.